lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
-    }
-    catch (error) {
-      this.kuzzle.pluginsManager.trigger('log:error', error);
-      throwOrStoreError(`Specification for the field ${indexName}.${collectionName}.${fieldName} triggered an error`, verboseErrors, errors);
+    if (Object.keys(fields).length > 0) {
+      return curateStructuredFields(this.typeAllowsChildren, fields, maxDepth);
     }
-  });
 
-  if (errors.length > 0) {
-    return {isValid: false, errors: errors};
+    return {};
   }
 
-  if (Object.keys(fields).length > 0) {
-    return curateStructuredFields(this.typeAllowsChildren, fields, maxDepth);
-  }
+  /**
+   * @param {FieldSpecification} fieldSpec
+   * @param {string} indexName
+   * @param {string} collectionName
+   * @param {string} fieldName
+   * @param {boolean} verboseErrors
+   * @returns {object}
+   * @throws InternalError
+   */
+  curateFieldSpecification(fieldSpec, indexName, collectionName, fieldName, verboseErrors) {
+    var
+      returnedTypeOptions,
+      result,
+      errors = [];
 
-  return {};
-};
+    verboseErrors = verboseErrors || false;
 
-/**
- * @param {FieldSpecification} fieldSpec
- * @param {string} indexName
- * @param {string} collectionName
- * @param {string} fieldName
- * @param {boolean} verboseErrors
- * @returns {object}
- * @throws InternalError
- */
-Validation.prototype.curateFieldSpecification = function validationCurateFieldSpecification (fieldSpec, indexName, collectionName, fieldName, verboseErrors) {
-  var
-    returnedTypeOptions,
-    result,
-    errors = [];
-
-  verboseErrors = verboseErrors || false;
+    result = this.curateFieldSpecificationFormat(fieldSpec, indexName, collectionName, fieldName, verboseErrors);
+    if (result.isValid === false) {
+      return result;
+    }
 
-  result = this.curateFieldSpecificationFormat(fieldSpec, indexName, collectionName, fieldName, verboseErrors);
-  if (result.isValid === false) {
-    return result;
-  }
+    _.defaultsDeep(fieldSpec, {
+      'mandatory': false,
+      'multivalued': {
+        value: false
+      }
+    });
 
-  _.defaultsDeep(fieldSpec, {
-    'mandatory': false,
-    'multivalued': {
-      value: false
+    if (!fieldSpec.hasOwnProperty('typeOptions')) {
+      fieldSpec.typeOptions = {};
     }
-  });
 
-  if (!fieldSpec.hasOwnProperty('typeOptions')) {
-    fieldSpec.typeOptions = {};
-  }
+    if (!checkAllowedProperties(fieldSpec.typeOptions, this.types[fieldSpec.type].allowedTypeOptions || [])) {
+      throwOrStoreError(`Field ${indexName}.${collectionName}.${fieldName} of type ${fieldSpec.type} is not specified properly`, verboseErrors, errors);
+    }
 
-  if (!checkAllowedProperties(fieldSpec.typeOptions, this.types[fieldSpec.type].allowedTypeOptions || [])) {
-    throwOrStoreError(`Field ${indexName}.${collectionName}.${fieldName} of type ${fieldSpec.type} is not specified properly`, verboseErrors, errors);
-  }
+    returnedTypeOptions = this.types[fieldSpec.type].validateFieldSpecification(fieldSpec.typeOptions);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 103 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)